src/components/Tags/__tests___/Tag.test.tsx

```diff
@@ -37,24 +31,34 @@ describe('Tag', () => {
       color: colors.solidFaintLight,
       fontSize: typography.fontSizeMini,
       fontFamily: typography.fontFamilySupport,
-      fontWeight: typography.fontWeightMedium.toString() as FontWeightType,
+      fontWeight: `${typography.fontWeightMedium}`,
     });
   });
 
-  it('renders correctly with custom props', () => {
+  it('renders tag correctly with an icon', () => {
+    const { getByTestId } = render(<Tag text="some tag" testID="TagID" icon="Clock" />);
+
+    const icon = getByTestId('TagID.Icon');
+
+    expect(icon).toBeTruthy();
+  });
+});
+
+describe.each(Object.keys(TAG_PROPERTIES))('renders correctly with custom variant %s', (variant) => {
```

Comment:
💅 It's not a custom variant, is predefined
```suggestion
describe.each(Object.keys(TAG_PROPERTIES))('With variant %s', (variant) => {
```

I think you can type the return of Object.keys here to avoid doing casts. Something like
```ts
const variants: TagVariant[] = Object.keys(TAG_PROPERTIES)...
```

Reply:
I tried to type the return of Object.keys as you suggested, but it gives the `Type 'string[]' cannot be assigned to type '("neutral" | "inversed" | "primary" | "alert" | "success" | "warning" | "disabled")[]'`.

Is there a specific way to approach this? 🤔

Reply:
Maybe because it doesn't have an `as const`, so the keys is infered as string

Reply:
Thank you ❤️ fixed in [6d26c52](https://github.com/magnetis/astro-native/pull/791/commits/6d26c5202fd3663b691c44c889a373d7f4ebd335)